### PR TITLE
Updates for component land model

### PR DESCRIPTION
**Pull request by uturuncoglu**
This PR aims to bring fully coupled atmosphere-land component configuration to UFS Weather Model. The newly added `sfc_land.[F90|meta]` is mainly responsible to receive information from component land model (NoahMP).

### CODE REVIEW

File: CODEOWNERS
Comment (grantfirl) [inline CODEOWNERS:199]: @uturuncoglu Since you're adding yourself to the CODEOWNERS, please accept the collaboration invitation for this repository that I just sent you so that you will be added as a reviewer automatically when these files are touched in a PR.

File: physics/noahmpdrv.meta
Comment (grantfirl) [inline physics/noahmpdrv.meta]: We've been moving to `do_*` rather than `flag_for*`, but since `flag_for_land_coupling` already exists and there are several others like it, it seems unfair to ask you to change it everywhere. This will have to be part of a separate standard name cleanup at some later point, I guess.

File: physics/sfc_land.f
Comment (grantfirl) [inline physics/sfc_land.f]: Since these are `intent(out)` they will be re-initialized with garbage values. If `cpllnd2atm` if false, then this routine will effectively return these garbage values. I don't know if that matters (if these variables are or ever will be used in the suite after this point when `cpllnd2atm` is false), but it might be safer to have these as `intent(inout)` so that they are not re-initialized.

Comment (grantfirl) [reply physics/sfc_land.f]: There is a CCPP rule that all `intent(out)` variables are initialized within the routine, even if it returns immediately. So, I think that the choices are to initialize them before returning (when `cpllnd2atm` if false) or change them to `intent(inout)`.

File: CODEOWNERS
Comment (uturuncoglu) [reply CODEOWNERS:199]: @grantfirl Sorry. I did not do it intentionally. I was just trying to follow the conventions in there. Since I am not doing to much development in CCPP side, you could remove me from the list. That is totally fine.

File: physics/noahmpdrv.meta
Comment (uturuncoglu) [reply physics/noahmpdrv.meta]: @grantfirl That is fine. I could fix them at least for land component coupling related once. When I look at to the examples for it, I found `do_mynnsfclay` but it's standard name is start with `flag_for`. Do you have any convention for it or just changing the argument name is fine at this point. Anyway, let me know what you think?

File: physics/sfc_land.f
Comment (uturuncoglu) [reply physics/sfc_land.f]: @grantfirl These variables are initialized under FV3/ccpp/data/GFS_typedefs.F90. So, I think it would be safe to change them to `intent(inout)`. Do you think that it would be safe to go with it?
Maybe this could be a discussion point in the CCPP Visioning Workshop and I think CCPP need to enforce those kind of standards and conventions in someway. Otherwise, it would be hard to keep the physics code in a certain quality.

File: CODEOWNERS
Comment (grantfirl) [reply CODEOWNERS:199]: @uturuncoglu IMO, it's totally appropriate to add you since you're adding these files to the repository. If @barlage is OK with being the CODEOWNER by himself, that is OK too. I'll let you two decide what you'd prefer.

File: physics/noahmpdrv.meta
Comment (grantfirl) [reply physics/noahmpdrv.meta]: Ya, I was talking about standard names only. I think that the variable names in the code are fine and the CCPP doesn't care. But, like I said, I wouldn't worry about changing it now since there is a lot of cleanup in other places to do anyway. The comment was more for the future, if you're going to be adding more on/off logicals, I think that having the standard names start with `do_X` is what the CCPP physics management committee prefers.

File: physics/sfc_land.f
Comment (grantfirl) [reply physics/sfc_land.f]: Ya, I think switching to `intent(inout)` is fine. The documentation (https://ccpp-techdoc.readthedocs.io/en/latest/CompliantPhysicsParams.html#coding-rules) mentions that "All intent(out) variables must be set inside the subroutine, including the mandatory variables errflg and errmsg.", but there is no automatic code-checking to enforce this. We've been relying on code reviews for this, but I agree, the gold standard would be to have the CCPP framework use a Fortran parser to check this.

File: physics/noahmpdrv.meta
Comment (uturuncoglu) [reply physics/noahmpdrv.meta]: @grantfirl Okay. I am not doing those changes at this point. Thanks.

File: physics/sfc_land.f
Comment (uturuncoglu) [reply physics/sfc_land.f]: @grantfirl i converted them from `out` to `inout`.

**Review by dustinswales — CHANGES_REQUESTED** [review]
@uturuncoglu I suggest changing the name of the new module, from sfc_land to something like noahmp_post?
Also, since this is a new module, is there a need to use fortran 77?

**Comment by uturuncoglu** [comment]
@dustinswales there is no need to use F77. I just follow the way with others similar such as sfc_ocean. In terms of changing name, I think it is better with this way since as I know there is other project to bring other external land component (GFDL's LM4) to the system. So, I prefer see the name generic to support also other developments in terms of land models. But, if you replay want to change it, I could do it easily. Let me know what you think?

**Comment by dustinswales** [comment]
@uturuncoglu We can leave the naming for another day. However, since this is new code, and it's a "ccpp-entrypoint", and there aren't many lines in this module, can you update to F90? (see https://ccpp-techdoc.readthedocs.io/en/v6.0.0/CompliantPhysicsParams.html#coding-rules)

**Comment by uturuncoglu** [comment]
@dustinswales Okay. I changed it to F90 and clean the module. Please check it again and let me know if you see anything else. Thanks.

**Review by dustinswales — APPROVED** [review]
Thanks for making these changes.

**Review by grantfirl — APPROVED** [review]

**Review by Qingfu-Liu — APPROVED** [review]

**Comment by jkbk2004** [comment]
Due to schedule issue of priority PRs, we are trying to combine https://github.com/ufs-community/ccpp-physics/pull/161 to this. A justification is that baseline cases changes are orthogonal: land vs rrfs case changes. @grantfirl Would it be ok? @uturuncoglu I will keep posting the decision.

**Comment by grantfirl** [comment]
> Due to schedule issue of priority PRs, we are trying to combine #161 to this. A justification is that baseline cases changes are orthogonal: land vs rrfs case changes. @grantfirl Would it be ok? @uturuncoglu I will keep posting the decision.

It's not ideal, but it OK with me.

**Comment by jkbk2004** [comment]
@uturuncoglu can you combine in https://github.com/ufs-community/ccpp-physics/pull/161 to this pr? @grantfirl thanks for the quick reply!

**Comment by uturuncoglu** [comment]
@jkbk2004 yes, that is fine for me.

**Comment by jkbk2004** [comment]
> @jkbk2004 yes, that is fine for me.

@uturuncoglu There are changes to combine at FV3 level as well: ccpp/data and ccpp/driver levels at https://github.com/NOAA-EMC/fv3atm/pull/773. If you feel it's not ideal, we can commit separately.

**Comment by uturuncoglu** [comment]
@jkbk2004 That is fine for me. As I see there is no ant conflict between these two PR. Right?

**Comment by jkbk2004** [comment]
> ccpp/data and ccpp/driver

@uturuncoglu I don't see major conflicts at the level of ccpp/data and ccpp/driver.

**Comment by uturuncoglu** [comment]
@jkbk2004 yes, me too. I think they could be merged together easily.

**Comment by grantfirl** [comment]
@uturuncoglu Let me know if you need any help merging #161 into your branch.

**Comment by uturuncoglu** [comment]
@grantfirl Okay. Let me merge it and test in my side quickly. BTW, Orion and Hercules are down. So, I'll use Derecho for testing.

**Comment by uturuncoglu** [comment]
@grantfirl @jkbk2004 Okay. I merged ccpp/physics one. Do I need to merge https://github.com/NOAA-EMC/fv3atm/pull/773 for the FV3 also.

**Comment by jkbk2004** [comment]
> @grantfirl @jkbk2004 Okay. I merged ccpp/physics one. Do I need to merge [NOAA-EMC/fv3atm#773](https://github.com/NOAA-EMC/fv3atm/pull/773) for the FV3 also.

@uturuncoglu Cool! Please, go ahead to combine https://github.com/NOAA-EMC/fv3atm/pull/773 to your https://github.com/NOAA-EMC/fv3atm/pull/682.

**Comment by uturuncoglu** [comment]
@jkbk2004 Did it. I am running land related RT on Derecho to be sure that are still working. It looks fine at this point.

**Comment by zach1221** [comment]
@grantfirl testing is complete one ufs-wm PR [#1845](https://github.com/ufs-community/ufs-weather-model/pull/1845). Can you merge this PR for us, please?

**Pull request merged**
